am-lib/src/test/java/integration/uk/gov/hmcts/reform/amlib/GetResourceDefinitionsWithCreatePermissionIntegrationTest.java

```diff
@@ -0,0 +1,143 @@
+
+        assertThat(result).containsExactly(ResourceDefinition.builder()
+            .resourceName(RESOURCE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .serviceName(SERVICE_NAME)
+            .build());
+    }
+
+    @Test
+    void shouldRetrieveResourceDefinitionWhenMultipleRecordsExistWithDifferentAttributes() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
+            JsonPointer.valueOf("/adult"), CREATE_PERMISSION));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
+            JsonPointer.valueOf("/child"), CREATE_PERMISSION));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).containsExactly(ResourceDefinition.builder()
+            .resourceName(RESOURCE_NAME)
+            .resourceType(RESOURCE_TYPE)
+            .serviceName(SERVICE_NAME)
+            .build());
+    }
+
+    @Test
+    void shouldReturnEmptyListWhenRecordExistsButNoCreatePermission() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(ROOT_ATTRIBUTE, READ_PERMISSION));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).isEmpty();
+    }
+
+    @Test
+    void shouldReturnEmptyListWhenNoRecords() {
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).isEmpty();
+    }
+
+    @Test
+    void shouldRetrieveMultipleResourceDefinitionsWhenMultipleRecordExistsForTheSameRole() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(CREATE_PERMISSION, resource, ROLE_NAME));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(CREATE_PERMISSION, resource2, ROLE_NAME));
+
+        List<ResourceDefinition> result =
+            service.getResourceDefinitionsWithCreatePermission(Collections.singleton(ROLE_NAME));
+
+        assertThat(result).containsExactly(resource2, resource);
+    }
+
+    @Test
+    void shouldRetrieveMultipleResourceDefinitionsWhenDefaultPermissionsExistForDifferentRoles() {
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(CREATE_PERMISSION, resource, ROLE_NAME));
+        importerService.grantDefaultPermission(createDefaultPermissionGrant(
+            CREATE_PERMISSION, resource2, OTHER_ROLE_NAME));
+
+        Set<String> userRoles = ImmutableSet.of(ROLE_NAME, OTHER_ROLE_NAME);
+
+        List<ResourceDefinition> result = service.getResourceDefinitionsWithCreatePermission(userRoles);
+
+        assertThat(result).containsExactly(resource, resource2);
+    }
+
+    private ResourceDefinition resourceBuilder(String resourceName) {
```

Comment:
Method should be renamed into `buildResource`. A `resourceBuilder` suggests that resource builder is returned while in fact final resource definition is returned.